src/sentry/api/serializers/models/group.py

```diff
@@ -447,38 +450,38 @@ def _get_status(self, attrs, obj):
             status_label = "unresolved"
         return status_details, status_label
 
-    def _get_permalink(self, obj, user):
+    def _is_authorized(self, user, organization_id):
         # If user is not logged in and member of the organization,
         # do not return the permalink which contains private information i.e. org name.
         request = env.request
-        is_superuser = request and is_active_superuser(request) and request.user == user
+        if request and is_active_superuser(request) and request.user == user:
+            return True
 
         # If user is a sentry_app then it's a proxy user meaning we can't do a org lookup via `get_orgs()`
         # because the user isn't an org member. Instead we can use the auth token and the installation
         # it's associated with to find out what organization the token has access to.
-        is_valid_sentryapp = False
         if (
             request
             and getattr(request.user, "is_sentry_app", False)
             and isinstance(request.auth, ApiToken)
         ):
-            is_valid_sentryapp = SentryAppInstallationToken.objects.has_organization_access(
-                request.auth, obj.organization
-            )
+            if SentryAppInstallationToken.objects.has_organization_access(
```

Comment:
The condition can now be simplified to:
```py
        if (
            request
            and (is_active_superuser(request) and request.user == user)
            or (
                # If user is a sentry_app then it's a proxy user meaning we can't do a org lookup via `get_orgs()`
                # because the user isn't an org member. Instead we can use the auth token and the installation
                # it's associated with to find out what organization the token has access to.
                getattr(request.user, "is_sentry_app", False)
                and isinstance(request.auth, ApiToken)
                and SentryAppInstallationToken.objects.has_organization_access(
                    request.auth, organization_id
                )
            )
        ):
            return True
```
This seems like overkill but I think you should merge up the `has_organization_access()` call.

Reply:
When you say to merge up `has_organization_access`, do you mean to lift its contents inline into this function?